sympy/core/exprtools.py

```diff
@@ -1316,22 +1316,6 @@ def _mask_nc(eq, name=None):
     >>> _mask_nc(eq, 'd')
     (A*_d0 + B*_d1, {_d0: Commutator(A, B), _d1: Commutator(A, C)}, [_d0, _d1, A, B])
 
-    If there is an object that:
-
-        - doesn't contain nc-symbols
-        - but has arguments which derive from Basic, not Expr
-        - and doesn't define an _eval_is_commutative routine
-
-    then it will give False (or None?) for the is_commutative test. Such
-    objects are also removed by this routine:
-
-    >>> from sympy import Basic
-    >>> eq = (1 + Mul(Basic(), Basic(), evaluate=False))
```

Comment:
@smichr this doctest failed because in this PR it is not possible to create a `Mul` with non-`Expr` args.

Can you think of another example where this would apply or is it okay just to remove this as unneeded now?

Maybe the accompanying function could also be simplified after this...